### PR TITLE
Local storage

### LINKED ISSUE #10: Add LocalStorage
## Value proposition

**As a** user,
**I want to** save my Favorites & added activities in LocalStorage
**so that** i can see my Favorites & added activities after refreshing the browser or if the browser was closed

## Acceptance criteria

- [x] after setting an activity as favorites, favorites will be saved in localstorage
- [x] added activities will be saved in localstorage

## Complexity

small/medium

## Tasks

- [x] create a branch
- [x] add localstorage
- [x] save favorites and (added) activities in localstorage

### PR DESCRIPTION
**Pull request by bensle**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 import db from './services/activityDB';
 
 export default function App() {
-  const [activities, setActivities] = useState(db);
+  const [activities, setActivities] = useLocalStorage('activities', db);
   const [typeFilterValue, setTypeFiltervalue] = useState('all');
   const [durationFilterValue, setDurationFilterValue] = useState('all');
   const [favorite, setFavorite] = useState([]);
   const [isHidden, setIsHidden] = useState(false);
+  const [savedFavorites, setSavedFavorites] = useLocalStorage('Favorites', []);
```
Comment (bensle) [inline src/App.js]: storage for Favorites

File: src/App.js [hunk 2]
```diff
@@ -29,16 +31,28 @@ export default function App() {
...
+  //   if (!favorite.includes(id)) setFavorite(favorite.concat(id));
+  // };
+
+  function baddToFavorites(id) {
+    const newFavorite = activities.find(acti => acti.id !== id);
     if (!favorite.includes(id)) setFavorite(favorite.concat(id));
-  };
+    setSavedFavorites([...savedFavorites, newFavorite]);
+    console.log(savedFavorites);
+  }
```
Comment (bensle) [inline src/App.js]: addFavorites to favoritesPage and set them to storageFavorites

File: src/App.js [hunk 3]
```diff
@@ -29,16 +31,28 @@ export default function App() {
...
 
   const removeFromFavorites = id => {
     const index = favorite.indexOf(id);
     const tempFavorites = [...favorite.slice(0, index), ...favorite.slice(index + 1)];
     setFavorite(tempFavorites);
   };
```
Comment (bensle) [inline src/App.js]: remove Favorites from FavPage/list

File: src/App.js [hunk 4]
```diff
@@ -29,16 +31,28 @@ export default function App() {
...
 
+  useEffect(
+    () => setSavedFavorites(() => activities.filter(activity => favorite.includes(activity.id))),
+    [activities, favorite] // eslint-disable-line react-hooks/exhaustive-deps
+  );
+
```
Comment (bensle) [inline src/App.js]: find activities for mapping and show on FavPage

File: src/hooks/useLocalStorage.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      if (typeof window !== 'undefined') {
+        window.localStorage.setItem(key, JSON.stringify(valueToStore));
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  return [storedValue, setValue];
+}
```
Comment (bensle) [inline src/hooks/useLocalStorage.js:34]: useLocalStorage hook

File: src/App.js [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
 import ActivityCards from './components/ActivityCard/ActivityCards';
 import DurationFilter from './components/DurationFilterButton/DurationFilter';
 import TypeFilter from './components/TypeFilterButton/TypeFilter';
+import useLocalStorage from './hooks/useLocalStorage';
 import FavoritesPage from './pages/FavoritesPage';
 import db from './services/activityDB';
 
 export default function App() {
-  const [activities, setActivities] = useState(db);
+  const [activities, setActivities] = useLocalStorage('activities', db);
```
Comment (bensle) [inline src/App.js:12]: activities in localstorage a re working find. New activities from form are also stored

File: src/App.js [hunk 7]
```diff
@@ -28,15 +28,18 @@ export default function App() {
...
-    const index = favorite.indexOf(id);
-    const tempFavorites = [...favorite.slice(0, index), ...favorite.slice(index + 1)];
-    setFavorite(tempFavorites);
+    const tempFavorites = [
+      ...activities.slice(0, index),
+      {...newFavorite, isFavorite: !newFavorite.isFavorite},
+      ...activities.slice(index + 1),
+    ];
+    setActivities(tempFavorites);
+    console.log('newFav', newFavorite);
```
Comment (macfrei) [inline src/App.js]:
```suggestion
```

**Review by macfrei — COMMENTED** [review]
Good job 🥳

**Pull request merged**